hivemind/moe/client/beam_search.py

```diff
@@ -306,6 +312,9 @@ async def _find_best_experts(
                     if isinstance(next_coord, int) and 0 <= next_coord < len(dim_scores)
                 ),
             )
+
+            if not best_active_pairs:
+                break
```

Comment:
If I understand everything right, then without this `if` we would also break the cycle, because eventually would get empty `beam` on L330 and reach `break` at L333. However on L332 we have `log.warning`, which telling us that this situation should get some attention.
TLDR : shouldn't we do `log.warning` here?

Reply:
Yes, I think that it should at least require an explanation via a comment/log entry

Reply:
good catch, this if is not necessary, removed

Reply:
Well, in fact this `if` gives some profit. We wont do a lot of calls just to find out that there is nothing to search for. As I understand, there will be 0 net communications in either way, be there will be some small time save.

Anyway, any change is acceptable: to add logging or to remove this `if` at all.